### ISSUE TITLE
Always true if condition mutator

### ISSUE THREAD

**Issue by manhunto** [infection/infection#1865]
Hi, I have idea to implement new mutator. Let's look at an example.

```php
final readonly class PaginationRequest
{
    public function __construct(
        private int $page,
        private int $size
    ) {
    }

    public function getSize(): int
    {
        return $this->size + 1;
    }

    public function getOffset(): int
    {
        if ($this->page === 1) {
            return 0;
        }

        return ($this->page - 1) * $this->size;
    }
}
```

And test for it
```php
class PaginationRequestTest extends TestCase
{
    /**
     * @dataProvider paginationInput
     */
    public function testItPreparesValidPaginationInput(int $page, int $expected): void
    {
        $sut = new PaginationRequest($page, 25);

        self::assertSame($expected, $sut->getOffset());
        self::assertSame(26, $sut->getSize());
    }

    /**
     * @return iterable<array{0: int, 1: int}>
     */
    public function paginationInput(): iterable
    {
        yield [1, 0];
        yield [2, 25];
        yield [3, 50];
        yield [10, 225];
    }
}
```

Link to playground: https://infection-php.dev/r/ylkd

In this example, if in getOffset() method is redundant and can be delated.
```php
        if ($this->page === 1) {
            return 0;
        }
```

What do you think about implementing a new mutator which will change if and elseif conditions to false. (and could be another mutator that set it to true)

This mutator will do changes like below
```diff
-        if ($this->page === 1) {
+        if (false) {
            return 0;
        }
```

If you think is worth implementing it, I can prepare PR with changes :)

**Comment by maks-rafalko** [comment]
In my todo list there was an idea to implement `if removal` mutator, so IMO removing any `if/elseif` conditions completely would be a great addition to the mutator set.

> and could be another mutator that set it to true

need to think more about it with some examples, probably worth trying to implement and check what it brings

**Comment by manhunto** [comment]
Yes, I also thought about `if/else removal`. What do you think is a better idea? Removal or setting it to false? IMO it would bring the same value. It seems to me that. `removing if` will be hard to implement.

I also thought about boolean conditions (`&&` and `||`) that are not in `if`/`elseif` statements. It could be done in the next PR. But I need to think about it or try to implement POC.

```diff
-$var = $a && $b;
+$var = false;
```

**Comment by manhunto** [comment]
> It seems to me that. removing if will be hard to implement.

Ok, I have to take those words back.. I thought that there is no empty node, but I found it in repo `Node\Stmt\Nop`. So let me know which approach, in your opinion, would be better ;)

**Comment by maks-rafalko** [comment]
let's try with changing condition to `false` then rather than removing. But feel free to change the strategy if removing will work better for some reason

**Comment by sanmai** [comment]
I'm afraid I'm going to upset you all. I think came up before, and the biggest argument against these mutations was that they are very easy to kill, very obvious from the coverage report, and most of the time they are going to waste CPU cycles, skewing the metrics.

Here are all mutations this code currently generates: https://infection-php.dev/r/3w26

**Comment by manhunto** [comment]
@sanmai you are right, I agree with you. Most of these mutation would not be useful but there will be a lot of it

**Comment by maks-rafalko** [comment]
@manhunto did you do some analysis and have examples?

**Comment by manhunto** [comment]
No, It is just my assumption and when I thought about it may be true. I will do more research after two weeks because I'm on holidays now
